projects/dsp-ui/src/lib/viewer/values/date-value/date-input-text/date-edit/date-edit.component.html

```diff
@@ -0,0 +1,39 @@
+<div [formGroup]="form">
+
+    <div class="era-radio">
+        <mat-radio-group class="era" [formControlName]="'era'" aria-label="Select an option" (change)="_handleInput()">
+            <mat-radio-button value="BCE">Before Common Era</mat-radio-button>
+            <mat-radio-button value="CE">Common Era</mat-radio-button>
+        </mat-radio-group>
+    </div>
+
+    <mat-form-field>
+        <mat-label>Year</mat-label>
+        <input class="year" matInput placeholder="Year" type="number" [formControlName]="'year'" min="1"
+               (input)="_handleInput()">
+        <mat-error *ngIf="yearControl.hasError('required')">
+            <span class="custom-error-message">A year is <strong>required</strong>.</span>
+        </mat-error>
+        <mat-error *ngIf="yearControl.hasError('min')">
+            <span class="custom-error-message">A valid year is greater than 0.</span>
+        </mat-error>
+    </mat-form-field>
+    <mat-form-field>
+        <mat-label>Month</mat-label>
+        <mat-select class="month" [formControlName]="'month'" (selectionChange)="_handleInput()">
```

Comment:
I don't really understand what you mean. Could we briefly talk about this?

Reply:
There would be more attractive ways to solve this in the UI but the main purpose of this PR was to provide a solution that works for dates of arbitrary precision and BCE and CE dates. As soon as the MatDatePicker supports this (if ever) it could be used instead.